src/components/course/routes/CourseAbout.jsx

```diff
@@ -28,6 +29,7 @@ const CourseAbout = () => {
 
   return (
     <>
+      <ExpiredSubscriptionModal />
```

Comment:
Wouldn't it be better to check the condition for rendering the `ExpiredSubscriptionModal` here instead of returning `null` from the component if condition fails? 

`{ customerAgreement?.hasCustomLicenseExpirationMessaging &&  <ExpiredSubscriptionModal /> }`

Reply:
Instead of using the subscription hook in the parent component and passing all props to the child component, I kept all the expired subscription-related functionality separate in that new component. Returning null from this component will not render anything on DOM.